components/resolved_ts/src/cmd.rs

```diff
@@ -0,0 +1,347 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use collections::HashMap;
+use kvproto::errorpb;
+use kvproto::raft_cmdpb::{AdminCmdType, CmdType, Request};
+use raftstore::coprocessor::{Cmd, CmdBatch};
+use raftstore::errors::Error as RaftStoreError;
+use tikv::server::raftkv::WriteBatchFlags;
+use txn_types::{Key, Lock, LockType, TimeStamp, Value, Write, WriteRef, WriteType};
+
+#[derive(Debug)]
+pub enum ChangeRow {
+    Prewrite {
+        key: Key,
+        lock: Lock,
+        value: Option<Value>,
```

Comment:
Now we have two fields that may contain the real value: the `short_value` in `lock` and this `value`. I prefer to avoid this. What about flattening the `Lock` struct and only keeping the necessary fields? Then we combine the short value and long value into one.
What do you think?

Reply:
And the same for `OnePc`

Reply:
I'm afraid of adding too many fields in the future. Keep both `short_value` and `value` is consistent with our original design, so I think it's ok.